### DISCUSSION TITLE
How to run a semantic skills from file

### DISCUSSION THREAD

**Discussion by sunnykim0011** [microsoft/semantic-kernel#4615] [Q&A] [answered]
Hi,
I'm running this **notebook**:
https://github.com/microsoft/semantic-kernel/blob/main/python/notebooks/02-running-prompts-from-file.ipynb

and this **function**:
result = jokeFunction("time travel to dinosaur age")
print(result)

How do I get a different response each time I call it?

How do I pass on $style variable when I call the function?

**Comment by moonbox3** [comment]
Hi @YoungKim01, thank you for your question. If you were to run the joke function over and over you may see a bit of a variation.

If you want to pass the `$style` variable in, you'd need to set up the `invoke_async` as follows:

```
context = kernel.create_new_context()
context["style"] = "Technical programming"
context["input"] = "Time travel to dinosaur age"
result = await jokeFunction.invoke_async(context=context)
print(result)
```

This would produce a result similar to:

```
Why did the programmer refuse to time travel to the dinosaur age?

Because they heard the T-Rex was a real "byte" out of the programming language!
```

**Comment by moonbox3** [comment, marked as answer]
This can now be done more easily via `KernelArguments` or as kwargs, as shown below. As of the beta, the context has been removed.

Method One:

`arguments = KernelArguments(style="technical programming", input="Time travel to dinosaur age")`

or Method Two:

`result = await kernel.invoke(jokeFunction, style="technical programming", input="Time travel to dinosaur age")`
